api/src/main/java/ca/bc/gov/educ/api/gradalgorithm/service/GradAlgorithmService.java

```diff
@@ -545,7 +546,7 @@ private void createCompleteGradMessage(StringBuilder currentGradMessage, Transcr
 			currentGradMessage.append(". ");
 		}
 
-		if(ruleProcessorData.getGradProgram().getProgramCode().contains("-PF") && dualDogwoodGraduated) {
+		if(ruleProcessorData.getGradProgram().getProgramCode().contains("-PF") && dualDogwoodGraduated && opMessage.equalsIgnoreCase(GRADUATED)) {
 			currentGradMessage.append("Student has successfully completed the Programme Francophone");
 			currentGradMessage.append(". ");
 		}
```

Comment:
Hi @sreepadkarekkad. Changes are ok, but while you are in here, there are two catch clauses that catch JsonProcessingExceptions that do no logging (line 176 and 123). Can you please add logging?